<?xml version="1.0"?>
<robot name="stm32_robot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  
  <!-- Include the robot structure -->
  <xacro:include filename="stm32_robot.urdf.xacro"/>
  
  <!-- Gazebo materials -->
  <gazebo reference="base_link">
    <material>Gazebo/Grey</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
  </gazebo>

  <gazebo reference="link_1">
    <material>Gazebo/Red</material>
  </gazebo>

  <gazebo reference="arm_1">
    <material>Gazebo/Green</material>
  </gazebo>

  <gazebo reference="link_2">
    <material>Gazebo/Red</material>
  </gazebo>

  <gazebo reference="arm_2">
    <material>Gazebo/Green</material>
  </gazebo>

  <gazebo reference="link_3">
    <material>Gazebo/Red</material>
  </gazebo>

  <gazebo reference="arm_3">
    <material>Gazebo/Green</material>
  </gazebo>

  <gazebo reference="end_effector">
    <material>Gazebo/Blue</material>
  </gazebo>

  <!-- ros2_control hardware interface -->
  <ros2_control name="STM32RobotSystem" type="system">
    <hardware>
      <plugin>gazebo_ros2_control/GazeboSystem</plugin>
    </hardware>
    
    <joint name="joint_1">
      <command_interface name="position">
        <param name="min">0</param>
        <param name="max">3.14159</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    
    <joint name="joint_2">
      <command_interface name="position">
        <param name="min">0</param>
        <param name="max">3.14159</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    
    <joint name="joint_3">
      <command_interface name="position">
        <param name="min">0</param>
        <param name="max">3.14159</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>

  <!-- Gazebo ros2_control plugin -->
  <gazebo>
    <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
      <parameters>$(find stm32_robot_controller)/config/controllers.yaml</parameters>
    </plugin>
  </gazebo>

</robot> 